core/include/log.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  core_util
+ * @{
+ *
+ * @file        log.h
+ * @brief       System logging header
+ *
+ * This header offers a bunch of "LOG_*" functions that, with the default
+ * implementation, behave like printf with verbosity levels.
+ *
+ * If desired, it is possible to implement a log module which then will be used
+ * instead the default printf-based implementation.  In order to do so, the log
+ * module has to provide "log_module.h", add -DHAVE_LOG_MODULE to CFLAGS and
+ * implement log_write().  See "sys/log/log_printfnoformat" for an example.
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef __LOG_H
+#define __LOG_H
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/**
+ * @brief defined log levels
+ *
+ * These are the logging levels a user can choose.
+ * The idea is to set LOG_LEVEL to one of these values in the application's Makefile.
+ * That will restrict output of log statements to those with equal or lower log level.
+ *
+ * The default log level is LOG_ALL, which will print every message.
```

Comment:
But, yes, I think that `LOG_INFO` makes most sense as a default, since >90% of RIOT users are currently RIOT developers and it still can be easily changed.